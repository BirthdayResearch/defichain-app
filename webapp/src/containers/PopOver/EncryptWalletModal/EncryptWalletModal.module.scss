// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../../scss/variables';
@import '../../../scss/bootstrap/variables';

.modalContent {
  background-color: #f4f3f6;
  border: none;
}

.modalBackdrop {
  background-color: #f4f3f6;
  opacity: 1 !important;
}

.lockIcon {
  width: 80px;
  height: 80px;
  background-color: $light;
  text-align: center;
  line-height: 80px;
  border-radius: 100%;
  margin-bottom: 20px;
  padding: 20px;
  color: $secondary;
}
