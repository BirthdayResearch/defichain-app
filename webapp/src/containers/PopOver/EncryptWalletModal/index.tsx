import React from 'react';
import { Modal, ModalBody } from 'reactstrap';
import { connect } from 'react-redux';
import styles from './EncryptWalletModal.module.scss';

import {
  closeEncryptWalletModal,
  encryptWalletFailure,
  encryptWalletStart,
} from '../reducer';
import EncryptWalletPage, {
  EncryptWalletPayload,
} from '../../WalletPage/components/EncryptWalletPage';

interface EncryptWalletModalProps {
  isWalletEncrypting: boolean;
  isEncryptWalletModalOpen: boolean;
  isErrorEncryptingWallet: string;
  encryptWalletFailure: (err: string) => void;
  encryptWalletStart: (item: EncryptWalletPayload) => void;
  closeEncryptWalletModal: () => void;
}

const EncryptWalletModal: React.FunctionComponent<EncryptWalletModalProps> = (
  props: EncryptWalletModalProps
) => {
  const {
    isEncryptWalletModalOpen,
    closeEncryptWalletModal,
    encryptWalletStart,
    isWalletEncrypting,
    isErrorEncryptingWallet,
    encryptWalletFailure,
  } = props;

  return (
    <Modal
      isOpen={isEncryptWalletModalOpen}
      contentClassName={styles.modalContent}
      backdropClassName={styles.modalBackdrop}
      style={{ width: '100%', maxWidth: '100vw' }}
      centered
    >
      <ModalBody>
        <EncryptWalletPage
          submitButtonLabel={'alerts.enableLocking'}
          onCloseFailure={encryptWalletFailure}
          pageErrorMessage={isErrorEncryptingWallet}
          pageLoadingMessage={
            'containers.wallet.encryptWalletPage.encryptingWallet'
          }
          isPageLoading={isWalletEncrypting}
          onSave={encryptWalletStart}
          onClose={closeEncryptWalletModal}
          isModal={true}
          pageSize={8}
        />
      </ModalBody>
    </Modal>
  );
};

const mapStateToProps = (state) => {
  const {
    isEncryptWalletModalOpen,
    isWalletEncrypting,
    isErrorEncryptingWallet,
  } = state.popover;
  return {
    isWalletEncrypting,
    isEncryptWalletModalOpen,
    isErrorEncryptingWallet,
  };
};

const mapDispatchToProps = {
  encryptWalletStart,
  encryptWalletFailure,
  closeEncryptWalletModal,
};

export default connect(mapStateToProps, mapDispatchToProps)(EncryptWalletModal);
