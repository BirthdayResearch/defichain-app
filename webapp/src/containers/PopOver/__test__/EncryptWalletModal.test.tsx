import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import store from '../../../app/rootStore';
import EncryptWalletModal from '../index';

describe('EncryptWalletModal', () => {
  it('should check for snapshot', () => {
    const wrapper = mount(
      <Provider store={store}>
        <EncryptWalletModal />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
