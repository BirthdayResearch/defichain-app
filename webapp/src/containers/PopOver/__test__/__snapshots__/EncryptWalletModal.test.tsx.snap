// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EncryptWalletModal should check for snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PopOver>
    <Connect(UpdateProgressModal)>
      <UpdateProgressModal
        dispatch={[Function]}
        isUpdateModalOpen={false}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={true}
          container="body"
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          trapFocus={false}
          unmountOnClose={true}
          zIndex={1050}
        />
      </UpdateProgressModal>
    </Connect(UpdateProgressModal)>
    <Connect(ReIndexModal)>
      <ReIndexModal
        closeReIndexModal={[Function]}
        isReIndexModelOpen={false}
        isRestartLoader={[Function]}
        reIndexMessage=""
        startSetNodeVersion={[Function]}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={true}
          container="body"
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          trapFocus={false}
          unmountOnClose={true}
          zIndex={1050}
        />
      </ReIndexModal>
    </Connect(ReIndexModal)>
    <Connect(BackupWalletNotice)>
      <BackupWalletNotice
        closeWalletDatBackupModal={[Function]}
        openBackupWalletDatModal={false}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={true}
          container="body"
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          trapFocus={false}
          unmountOnClose={true}
          zIndex={1050}
        />
      </BackupWalletNotice>
    </Connect(BackupWalletNotice)>
    <Connect(ResetWalletDatModal)>
      <ResetWalletDatModal
        closeResetWalletDatModal={[Function]}
        openResetWalletDatModal={false}
        startResetWalletDatRequest={[Function]}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={
            Object {
              "mountOnEnter": true,
              "timeout": 150,
            }
          }
          centered={true}
          container="body"
          fade={true}
          isOpen={false}
          keyboard={true}
          modalTransition={
            Object {
              "timeout": 300,
            }
          }
          onClosed={[Function]}
          onOpened={[Function]}
          returnFocusAfterClose={true}
          role="dialog"
          scrollable={false}
          trapFocus={false}
          unmountOnClose={true}
          zIndex={1050}
        />
      </ResetWalletDatModal>
    </Connect(ResetWalletDatModal)>
  </PopOver>
</Provider>
`;
